### ISSUE TITLE
What is the encouraged way to detect a window focus loss?

### ISSUE THREAD

**Issue by SpoinkyNL** [canton7/Stylet#90]
Hello,

I'm trying to detect whether my root view has lost focus.  

So I thought this would trigger whenever a focus loss occurs:

```cs
protected override void OnDeactivate()
{
    Debug.WriteLine("I've lost focus in OnDeactive'");
    base.OnDeactivate();
}
```

But unfortunately this only gets called upon a window **minimize**. A focus loss doesn't trigger OnDeactivate.

So what I wrote instead, just to see whether it even should be triggered upon a focus loss is this:
```cs
protected override void OnViewLoaded()
{
    ((RootView)View).Deactivated += (sender, args) =>
    {
        Debug.WriteLine("I've lost focus in Deactivated'");
    };
    base.OnViewLoaded();
}
```

That works, but it's not very MVVM, I can call OnDeactivate() from there so that conductors trigger OnActivate() on their items etc. etc., but then I'm also responsible for calling OnActivate().

Also this might be undesirable since minimize guarantees the user can't see the window whereas a focus loss does not. (For me the focus loss is a requirement), so what's the suggested route? :)

Oh, and thanks for your work, I've been enjoying Stylet

**Comment by canton7** [comment]
You can use Stylet's Actions with events.

```xaml
<Window ... Deactivated="{s:Action WindowDeactivated}">
```

Then define a `WindowDeactivated` method in your VM.

**Comment by SpoinkyNL** [comment]
Hey, thanks for the quick reply

That'll work for me. But to let child VMs know about the focus change I'm best off with the EventAggregator?

**Comment by canton7** [comment]
Probably. It might make sense to replace the WindowConductor with something of your own, which activates/deactivates the main window VM based on focus state, but it's a little hard to tap into that at the moment: you'll have to subclass the WindowManager.

**Comment by SpoinkyNL** [comment]
Thanks, I'll go with the EventAggregator

**Comment by canton7** [comment]
If take a PR which let's you replace the WindowConductor a bit more easily, but I'm afraid I don't have much time to do it at the moment.
